docs/loadbalancing.md

```diff
@@ -0,0 +1,120 @@
+# Layer 4 and Layer 7 Load Balancing with KubeLB
+
+## Layer 4
+
+KubeLB supports Layer 4 load balancing by leveraging the Kubernetes Service abstraction. When a Service of type `LoadBalancer` is created in the tenant cluster, KubeLB CCM will create a new `LoadBalancer` resource in the LB cluster. This resource will be responsible for creating the necessary network resources to expose the Service to the internet.
+
+The `LoadBalancer` resource created by KubeLB CCM is a Kubernetes Custom Resource Definition (CRD) that is managed by the KubeLB controller. The controller watches for changes to the `LoadBalancer` resources and performs the following actions:
+
+- Creates a corresponding service of type loadbalancer in the LB cluster.
+- Propagates the load balancing configuration to the envoy proxy to route traffic to the backend pods.
+
+Example of a Service of type `LoadBalancer`:
+
+```yaml
+apiVersion: kubelb.k8c.io/v1alpha1
+kind: LoadBalancer
+metadata:
+  name: loadbalancer
+spec:
+  endpoints:
+    - addresses:
+        - ip: 168.119.189.211
+        - ip: 168.119.185.115
+      ports:
+        - name: a-port
+          port: 32019
+          protocol: TCP
+        - name: b-port
+          port: 32529
+          protocol: TCP
+  ports:
+    - name: a-port
+      port: 8080
+      protocol: TCP
+    - name: b-port
+      port: 8081
+      protocol: TCP
+```
+
+## Layer 7
+
+While Layer 4 load balancing is sufficient for many use cases, Layer 7 load balancing provides additional features such as path-based routing, header-based routing, and SSL termination.
+
+Since Layer 7 load balancing is not natively supported by Kubernetes, KubeLB uses the Ingress and Gateway API resources to provide Layer 7 load balancing capabilities. For l4, we could simply rely on a single resource i.e. service of type LoadBalancer. However, for l7, we need to rely on multiple resources to achieve the desired functionality. This list of resources is also going to be extended in the future as we add more features to KubeLB.
+
+Eventually the workflow for Layer 7 load balancing will look like this:
+
+- Resources for Application load balancing are created in the tenant cluster.
+- KubeLB CCM watches for changes to the Ingress and Gateway resources in the tenant cluster.
+- KubeLB CCM creates corresponding resources in the LB cluster to configure the load balancer.
+
+For transmitting these resources in the LB cluster, we had the following options:
+
+1. Use the original Ingress and Gateway resources.
+2. KubeLB manager uses kubeconfigs to access the tenant cluster and create the resources in the LB cluster.
+3. Use the same CRD `LoadBalancer` as Layer 4 and extend it to support Layer 7.
+4. Create a new CRD called `Route` to cover routing for both L4 and L7 flows.
+
+### Use the original Ingress and Gateway resources
```

Comment:
As we discussed it internally, this was my preferred choice, but given the exposure topic, I would second the creation of the new Route CRD, the only concern is not to overcomplicate its structure.